### ISSUE TITLE
📝 Unusual formatter output when a JSX node is followed by a text node starting with a single-character-word

### ISSUE THREAD

**Issue by simon-paris** [biomejs/biome#1885]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.8.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/1.22.18"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{}
```

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIABSAGUAYQBjAHQAIABmAHIAbwBtACAAIgByAGUAYQBjAHQAIgA7AAoACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAEEAcABwACgAKQAgAHsACgAJAC8ALwAgAFQAaABpAHMAIABpAHMAIABmAG8AcgBtAGEAdAB0AGUAZAAgAGQAaQBmAGYAZQByAGUAbgB0AGwAeQAgAGkAbgAgAHAAcgBlAHQAdABpAGUAcgAgAGEAbgBkACAAYgBpAG8AbQBlACwAIABhAG4AZAAgAEkAIAB0AGgAaQBuAGsAIABiAG8AdABoACAAYQByAGUAIAB3AHIAbwBuAGcALgAKAAkALwAvACAASQB0ACAAaABhAHAAcABlAG4AcwAgAG8AbgBsAHkAIAB3AGgAZQBuACAAdABoAGUAIABmAGkAcgBzAHQAIAB3AG8AcgBkACAAbwBmACAAdABoAGUAIAB0AGUAeAB0ACAAbgBvAGQAZQAgAGgAYQBzACAAbABlAG4AZwB0AGgAPQAxAC4AIABJACAAdABoAGkAbgBrACAAdABoAGUAIABjAG8AcgByAGUAYwB0ACAAbwB1AHQAcAB1AHQAIABzAGgAbwB1AGwAZAAgAGIAZQAgAHQAbwAgAGwAZQBhAHYAZQAgAHQAaABpAHMAIAB1AG4AYwBoAGEAbgBnAGUAZAAsACAAdwBoAGkAYwBoACAAaQBzACAAdwBoAGEAdAAgAGgAYQBwAHAAZQBuAHMAIABpAGYAIAB0AGgAZQAgAGYAaQByAHMAdAAgAHcAbwByAGQAIABpAHMAIABuAG8AdAAgAGwAZQBuAGcAdABoAD0AMQAuAAoACQByAGUAdAB1AHIAbgAgACgACgAJAAkAPAA%2BAAoACQAJAAkAPABzAHAAYQBuACAALwA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAegAgAGoAcwB4ACAAbgBvAGQAZQBzAAoACQAJAAkAPABzAHAAYQBuACAALwA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAegAgAGYAbwBsAGwAbwB3AGUAZAAgAGIAeQAgAHQAZQB4AHQAIABuAG8AZABlAHMACgAJAAkACQA8AHMAcABhAG4AIAAvAD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB6ACAAdwBoAGkAYwBoACAAcwB0AGEAcgB0ACAAdwBpAHQAaAAgAGEAIABzAGkAbgBnAGwAZQAgAGMAaABhAHIAYQBjAHQAZQByAAoACQAJADwALwA%2BAAoACQApADsACgB9AAoA

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by fireairforce** [comment]
It seems that prettier allows any single character word to hug the following or previous line. This is a normal scene, please refer to pr: https://github.com/biomejs/biome/pull/1075, i am not sure this rule cause u a bug?

**Comment by simon-paris** [comment]
I understand the special case for not inserting a line after a punctuation mark, but surely it should not forcibly remove a line because the next sentence starts with the word "I" or "A".

Here's the actual code that triggered this issue for me.

```
<Text color={theme.urbanGrey60} size="xs">
  Mudah.my
  <br />
  Sdn Bhd [200701024583 (782603-V)]
  <br />
  a Carousell Group company
  <br />
</Text>
```

**Comment by arendjr** [comment]
I agree with @simon-paris here. It looks like the PR that is being referred to implemented the change in order to increase Prettier compatibility, but I also think the old Biome behavior made more sense.

**Comment by ematipico** [comment]
Maybe we could mark this as a divergence. I also feel like that Biome's behaviour makes more sense

**Comment by ematipico** [comment]
Closing because Prettier behaviour isn't idempotent.
